astropy/coordinates/sky_coordinate.py

```diff
@@ -1132,6 +1132,18 @@ def is_equivalent_frame(self, other):
                 "Tried to do is_equivalent_frame on something that isn't frame-like"
             )
 
+    def _prepare_for_separation(self, other):
+        if not self.is_equivalent_frame(other):
+            kwargs = {"merge_attributes": False} if isinstance(other, SkyCoord) else {}
+            try:
+                other = other.transform_to(self, **kwargs)
+            except TypeError:
+                raise TypeError(
```

Comment:
while we're at it maybe we could also chain these exceptions either as

```python
try:
    ...
except TypeError as exc:
    raise TypeError(...) from exc
```
or
```python
try:
    ...
except TypeError:
    raise TypeError(...) from None
```
? (I think the second one is more appropriate here)

Reply:
It doesn't look like this code path can ever be triggered. Trying to compute the separation of a frame without data raises a `ValueError`, not a `TypeError`:
```python
>>> from astropy.coordinates import FK5, SkyCoord
>>> SkyCoord(1, 1, unit="deg").separation(FK5())
Traceback (most recent call last):
  ...
ValueError: Cannot transform a frame with no data
```
To me it seems that this error message is clear enough, so this `try-except` block can be removed.